test/unit/services/AuthServiceTests.ts

```diff
@@ -0,0 +1,128 @@
+import axios from "axios";
+import {assert, expect} from 'chai';
+import type { LoginRequest } from "../../../src/models/LoginRequest"
+import MockAdapter from "axios-mock-adapter";
+import { getToken } from "../../../src/services/AuthService";
+
+const mock = new MockAdapter(axios);
+
+const URL = "http://localhost:8080/api/auth/login";
+
+describe('AuthService', function () {
+    describe('getToken', function () {
+        it('should return token when valid request', async () => {
+            const loginRequest: LoginRequest = {
+                email: "admin@kainos.com",
+                password: "admin"
+            }
+            mock.onPost(URL, loginRequest).reply(200, "Correct");
+
+            try {
+                await getToken(loginRequest);
+
+            } catch (e) {
+                assert.fail("Expected no error message");
+            }
+        }),
+        it('should return error when email invalid format', async () => {
+            const loginRequest: LoginRequest = {
```

Comment:
Like in the back-end these can be parameterised tests. Below is an example in Typescript - first the method
```
export const isPerfectSquare = function(value: number): boolean {
    if (value < 0) {
        return false;
    }
    const sqrt = Math.floor(Math.sqrt(value));

    return sqrt * sqrt === value;
}
```

and here is the test for the method.

```
describe('isPerfectSquare', function () {
    const testCases: [number, boolean][] = [
      [16, true],
      [14, false],
      [25, true],
      [0, true],
      [1, true],
      [-4, false],
      [20, false]
    ];

    testCases.forEach(([input, expectedResult]) => {
      it(`will return ${expectedResult} when input is ${input}`, function() {
        const actualResult = isPerfectSquare(input);
        expect(actualResult).to.equal(expectedResult);
      })
    });
  });

```

Reply:
Added, tested, and committed.